backend/de.metas.async/src/main/java/de/metas/async/api/IAsyncBatchBL.java

```diff
@@ -110,7 +113,17 @@
 			@NonNull List<T> model,
 			@NonNull String asyncBatchInternalName);
 
+	IAutoCloseable assignTempAsyncBatchIdToModel(@NonNull Object model, @NonNull AsyncBatchId asyncBatchId);
+
 	I_C_Async_Batch getAsyncBatchById(AsyncBatchId asyncBatchId);
 
 	AsyncBatchId newAsyncBatch(String asyncBatchType);
+
+	Optional<String> getAsyncBatchTypeInternalName(@NonNull final I_C_Async_Batch asyncBatch);
+
+	boolean isAsyncBatchTypeInternalName(@NonNull I_C_Async_Batch asyncBatch, @NonNull String expectedInternalName);
+
+	Optional<AsyncBatchType> getAsyncBatchType(@NonNull I_C_Async_Batch asyncBatch);
```

Comment:
```suggestion
```

can be done later: 	I think also this one is not needed. instead we can use `getAsyncBatchTypeById(@NonNull AsyncBatchTypeId asyncBatchTypeId);`

Reply:
is for convenience. the logic is used in couple of places.